frontend/pages/queries/edit/components/EditQueryForm/EditQueryForm.tsx

```diff
@@ -188,6 +194,37 @@ const EditQueryForm = ({
   const [isSaveAsNewLoading, setIsSaveAsNewLoading] = useState(false);
   const [showAdvancedOptions, setShowAdvancedOptions] = useState(false);
   const [queryWasChanged, setQueryWasChanged] = useState(false);
+  const [selectedTargetType, setSelectedTargetType] = useState("");
+  const [selectedLabels, setSelectedLabels] = useState({});
+
+  useEffect(() => {
```

Comment:
Is this effect necessary? Would setting `storedQuery`'s values as the defaults for above states not
work? Any updates to `storedQuery` would cause the component to rerender I'd think?

Reply:
When this initially renders, the stored query isn't loaded yet so we can't use it as the default.  But if we pull the `set*` calls out of `useEffect`, they will trigger a re-render every time they're called, so we end up in an infinite render loop.  It's the fact that the state vars aren't populated directly from `storedQuery`, but from values _calculated_ from `storedQuery`, that make this tricky.  There may be a better solution than `useEffect()`, but it definitely does the trick.